### DISCUSSION TITLE
Show legacy names

### DISCUSSION THREAD

**Discussion by LumiMakesStuff** [godotengine/godot-proposals#10403] [Engine Core]
I think we should be able to add objects and use them in code with their legacy names.
Reasoning: Many tutorials use KinematicBody2D which was renamed to CharacterBody2D and newer users may not be able to figure that out.

**Comment by LumiMakesStuff** [comment]
And I mean the nodes
can it also have code support

**Comment by theraot** [comment]
The renamed classes also had their methods changed, for example `move_and_slide` no longer takes arguments.

Also, It is not just renames. For example some classes don't have a direct replacement.

Supporting the old API on top of the new one would: 1. Make maintaining Godot harder. 2. Make Godot and its games bigger for no added features.

Letting Godot break compatibility allows Godot to improve faster.

---

If the problem is that some class is not available, and the developer is not capable of figuring out why on their own... It is relatively trivial to find out by using the forums, or discord, or some Q&A site.

On the flip side, if these old classes appear in newer versions of Godot, but with slightly different API or behaviour, for the people trying to help on said forums, discord, and Q&A sites, it would be much harder to piece out what is going on. Why is the class not working the way it works in the tutorial? It would not be evident that they are not using the version of Godot the tutorial is intended to (many people asking don't say what version of Godot they are using, not share what tutorial they are following).

Ultimately, developers should learn. In fact, if they are looking for tutorials, they are already trying to learn. And figuring out what version of Godot the tutorial is for, is another lesson to learn.

---

Sadly there are many tutorials for Godot 3 that do not say they are for Godot 3 (because Godot 2 wasn't ever as popular), At least - I believe - Godot 4 tutorials are more forward in telling the users for what version they are for.

We need is more people making more tutorials... As Godot 4 is becoming even more popular than Godot 3. This problem might go away on its own once finding Godot 3 is relatively hard, and Godot 4 tutorials are abundant.

---

I don't have a ton of followers anywhere. And for me it is hard to justify dedicating time to make tutorials, as I'm the sole developer on my organization at the time of writing, I should be programming... Yet, I have answered some questions regarding the transition from Godot 3 to Godot 4, I have even explained how to tell apart Godot 3 tutorials and Godot 4 tutorials: https://stackoverflow.com/a/75870815/402022 - Let more people make tutorials about it.

The more authors make tutorials about this, the greater the chance to learn this is for those developers who are not capable of figuring out what version a tutorial is for, whose main form of learning is tutorials and cannot use forums, discord, or Q&A sites.

---

I'd wish the version number was in the default splash screen so nobody missed it. The version number is always in the bottom of the screen on the editor, but not everybody knows to look there, and some tutorials would crop it out. I mentioned that here:

- https://github.com/godotengine/godot-proposals/issues/7243

I'd wish the "lagacy" names where in the description of classes, and that the "Create New Node" dialog were able to search in the description text. *This could be a new proposal.*

**Reply by LumiMakesStuff** [reply]
Ah ok i understand. Thanks
